cpp/src/parquet/column_writer.cc

```diff
@@ -854,8 +858,21 @@ void ColumnWriterImpl::AddDataPage() {
   InitSinks();
   num_buffered_values_ = 0;
   num_buffered_encoded_values_ = 0;
+  num_buffered_rows_ = 0;
 }
 
+namespace {
+
+int32_t safe_cast_64_to_32(int64_t value) {
+  if (value > std::numeric_limits<int32_t>::max() ||
+      value < std::numeric_limits<int32_t>::min()) {
+    throw ParquetException("Cannot cast ", value, " to int32_t");
+  }
+  return static_cast<int32_t>(value);
+}
```

Comment:
Is this necessary? Is it even possible to write more than `std::numeric_limits<int32_t>::max()` values to a row group in parquet?

Reply:
I have removed the check. Please take a look again. Thanks @wjones127